src/protocols/filter/components/block_filters_process.rs

```diff
@@ -95,6 +95,28 @@ impl<'a> BlockFiltersProcess<'a> {
                         self.peer
                     );
                 } else {
+                    // if the only matched block is the prove state block, then request block data directly
+                    possible_match_blocks
+                        .retain(|block_hash| block_hash != &prove_state_block_hash);
+                    if possible_match_blocks.is_empty() {
```

Comment:
```suggestion
                    if possible_match_blocks.contains(&prove_state_block_hash) {
```
Otherwise, it will missing the `prove_state_block_hash` block when there are multiple hashes in `possible_match_blocks`.